src/api/article/author/controller.js

```diff
@@ -0,0 +1,39 @@
+import pick from 'lodash/pick';
+
+import { sendJson } from 'utils/api';
+
+import User, { serializeAuthor } from 'api/user/model';
+
+export const getAll = (req, res, next) =>
+  User.find({ active: true, role: 'author' })
+    .then(users => users.map(serializeAuthor))
+    .then(sendJson(res))
+    .catch(next);
+
+export const create = async ({ body }, res, next) => {
+  const emailRgxp = /^generated-author-\d+@wir.by$/;
+  const capturingRgxp = /^generated-author-(.*?)@wir.by$/;
+
+  let nextNum = 0;
+  await User.find({ email: { $regex: emailRgxp } }).then(authors => {
+    if (authors.length > 0) {
+      nextNum =
+        authors
+          .map(({ email }) => parseInt(capturingRgxp.exec(email)[1], 10))
```

Comment:
`capturingRgxp.exec(email)[1]` - bad readability.

Mb we should use `es2018` [new regex feature](https://twitter.com/housecor/status/960153700206510082?ref_src=twcamp%5Eshare%7Ctwsrc%5Eios%7Ctwgr%5Eorg.telegram.TelegramHD.Share%7Ctwcon%5E7100%7Ctwterm%5E0)?

Should work with [this babel plugin](https://github.com/dmitrysoshnikov/babel-plugin-transform-modern-regexp) :)

Reply:
I made some research: to enable the feature we need this babel plugin + `regexp-tree-runtime` workaround as named groups are not fully implemented yet in the plugin: https://github.com/dmitrysoshnikov/babel-plugin-transform-modern-regexp#useruntime-option

I would like to keep numbered group instead, they are read pretty straightforward. I've also replaced two regexps with one and moved to the `model` file.